app/api/route/handler/user_handler.py

```diff
@@ -66,14 +32,21 @@ def create_user(api_context: ApiContext) -> UserResponse:
 
         user.roles = user_roles
 
-    return UserResponse.from_orm(user)
+    api_context.db_session.flush()
+    return UserOut(**UserOut.Schema().dump(user))
 
 
-def patch_user(user_id: str, api_context: ApiContext) -> UserResponse:
+def patch_user(user_id: str, request: UserPatchParams, api_context: ApiContext) -> UserOut:
     user = get_or_404(api_context.db_session, User, user_id)
 
-    request = UserPatchParams.parse_obj(api_context.request_body)
-    for key, value in request.get_set_params():
+    for key, value in request.get_set_params(api_context):
+        # Unfortunately there isn't a great way of determining
+        # which values were explicitly set in the request other
+        # than to look at the raw request JSON. Because the objects
+        # get converted to dataclasses, all the values are "set" and
+        # dataclasses don't keep track of the parameters
+        if key not in api_context.request_body:
+            continue
 
         if key == "roles":
             handle_role_patch(user, value, api_context)
```

Comment:
The way patch works for a user is already a bit iffy. Roles are handled, but it has a lot of complexity (see `handle_role_patch`). Based on what I've read online, patch endpoints probably make sense to split up for anything dealing with one-to-many relationships. Roles should probably be handled by endpoints like `PATCH /user/:id/add-role` and `PATCH /user/:id/remove-role/:role_id` rather than this way. Those would work significantly more intuitively and avoid all of the complexity. It just leads to having a lot more endpoints potentially.

Reply:
agree that patching roles this way is iffy. shall we remove the patching of roles from the template for now then?